### DISCUSSION TITLE
Filter query by column in joined table

### DISCUSSION THREAD

**Discussion by kevinmichaelchen** [aarondl/sqlboiler#891] [Q&A] [answered]
I have a simple data model: dogs and humans.
A dog belongs to a human (i.e., dog has the foreign key).
I want to make a query that filters on a column in the table to which I'm joining.

```sql
SELECT h.name, d.name
  FROM dogs d
  JOIN humans h ON h.id = d.owner_id
  WHERE h.name = 'Kevin'
```

I'm wondering if this is possible using eager loading, but ultimately eager loading is not necessary.

Ideally, I'd like to avoid raw SQL, and I'd like to avoid writing out all columns in my query.

**Comment by kevinmichaelchen** [comment]
Here's what I've come up with:
```go
dogs, err := models.Dogs(
  // eager load the dog's hooman
  qm.Load(models.DogRels.Human, 
    qm.Where(fmt.Sprintf("%s.%s = ?", models.TableNames.Humans, models.HumanColumns.Name), humanNames[0]),
  ),
  qm.Limit(5),
).All(ctx, exec)
```

The `dogs` slice will sometimes have an entity with a nil Human (if for example there are fewer than `limit` dogs that meet the `where` criteria).

**Comment by kevinmichaelchen** [comment, marked as answer]
Using an `InnerJoin` means we filter out the records that don't match our `WHERE` criteria.
```go
func queryForDogs() {
	dogs, err := models.Dogs(
		// eager load the dog's hooman
		qm.Load(models.DogRels.Human),
		qm.InnerJoin(
			fmt.Sprintf(
				"%s on %s.%s = %s.%s",
				models.TableNames.Humans,
				models.TableNames.Humans,
				models.HumanColumns.ID,
				models.TableNames.Dogs,
				models.DogColumns.HumanID,
			),
		),
		qm.Where(
			fmt.Sprintf(
				"%s.%s = ?", 
				models.TableNames.Humans,
				models.HumanColumns.Name,
			),
			humanNames[0],
		),
		qm.Limit(5),
	).All(ctx, exec)
}
```

**Reply by firminochangani** [reply]
Thanks @kevinmichaelchen!!!
